packages/flex-plugin-scripts/src/scripts/__tests__/start.test.ts

```diff
@@ -118,26 +171,102 @@ describe('StartScript', () => {
-      const writeFileSync = jest
-        .spyOn(fs.default, 'writeFileSync')
-        .mockReturnValue();
+      const writeJSONFile = jest
+        .spyOn(fs, 'writeJSONFile')
+        .mockReturnThis();
 
       startScripts._updatePluginsUrl(port);
 
       expect(requirePackages).toHaveBeenCalledTimes(1);
-      expect(writeFileSync).toHaveBeenCalledTimes(1);
+      expect(writeJSONFile).toHaveBeenCalledTimes(1);
       expect(plugins[0].src.indexOf(port.toString()) !== -1).toBeTruthy();
 
       requirePackages.mockRestore();
-      writeFileSync.mockRestore();
+      writeJSONFile.mockRestore();
+    });
+  });
+
+  describe('updatePluginsPort', () => {
+    const port = 1234;
+    const plugin = { name: 'plugin-test', dir: 'test-dir', port: 0 };
+    const readPluginsJson = jest
+      .spyOn(fs, 'readPluginsJson');
+    const writeJSONFile = jest
+      .spyOn(fs, 'writeJSONFile');
+    const _parseUserInputPlugins = jest
+      .spyOn(startScripts, '_parseUserInputPlugins');
+
+    beforeEach(() => {
+      _parseUserInputPlugins.mockReturnValue([{name: 'plugin-test', remote: false}]);
+      readPluginsJson.mockReturnValue({plugins: [plugin]});
+      writeJSONFile.mockReturnThis();
+    });
+
+    afterAll(() => {
+      readPluginsJson.mockRestore();
+      _parseUserInputPlugins.mockRestore();
+      writeJSONFile.mockRestore();
+    });
+
+    it('should update the plugin port', () => {
+      startScripts._updatePluginPort(port, plugin.name);
+
+      expect(writeJSONFile).toHaveBeenCalledTimes(1);
+      expect(writeJSONFile).toHaveBeenCalledWith(CliPath, {'plugins': [ {...plugin, port}]});
+    });
+
+    it('should not update the plugin port', () => {
+      startScripts._updatePluginPort(port, 'unknown-plugin');
+
+      expect(writeJSONFile).toHaveBeenCalledTimes(1);
+      expect(writeJSONFile).toHaveBeenCalledWith(CliPath, {'plugins': [ {...plugin}]});
+    });
+  });
+
+  describe('_parseUserInputPlugins', () => {
+    const plugin = { name: 'plugin-test', dir: 'test-dir', port: 0 };
+    const readPluginsJson = jest
+      .spyOn(fs, 'readPluginsJson');
+
+    beforeEach(() => {
+      readPluginsJson.mockReturnValue({plugins: [plugin]});
+    });
+
+    afterEach(() => {
+      readPluginsJson.mockRestore();
+    });
+
+    it('should return the local plugins only if found', () => {
+      const result = startScripts._parseUserInputPlugins(...['--name', 'plugin-test']);
+
+      expect(readPluginsJson).toHaveBeenCalledTimes(1);
+      expect(result).toEqual([{'name': 'plugin-test', 'remote': false}]);
+    });
+
+    it('should always return the remote plugins', () => {
+      const result = startScripts._parseUserInputPlugins(...['--name', 'plugin-test@remote']);
+
+      expect(readPluginsJson).toHaveBeenCalledTimes(1);
+      expect(result).toEqual([{'name': 'plugin-test', 'remote': true}]);
+    });
+
+    it('should throw an error if local plugin is not found', (done) => {
```

Comment:
Add a test for the unexpected name format https://github.com/twilio/flex-plugin-builder/pull/288/files#diff-86791f499dd282bf940eeb1ff21014faR217 too